app/src/main/java/org/wikiedufoundation/wikiedudashboard/di/Modules.kt

```diff
@@ -0,0 +1,339 @@
+import org.wikiedufoundation.wikiedudashboard.ui.courselist.view.CourseListView
+import org.wikiedufoundation.wikiedudashboard.ui.dashboard.MyDashboardContract
+import org.wikiedufoundation.wikiedudashboard.ui.dashboard.MyDashboardPresenterImpl
+import org.wikiedufoundation.wikiedudashboard.ui.dashboard.RetrofitMyDashboardProvider
+import org.wikiedufoundation.wikiedudashboard.ui.mediadetail.MediaDetailsContract
+import org.wikiedufoundation.wikiedudashboard.ui.mediadetail.MediaDetailsPresenterImpl
+import org.wikiedufoundation.wikiedudashboard.ui.mediadetail.RetrofitMediaDetailsProvider
+import org.wikiedufoundation.wikiedudashboard.ui.profile.ProfileContract
+import org.wikiedufoundation.wikiedudashboard.ui.profile.ProfilePresenterImpl
+import org.wikiedufoundation.wikiedudashboard.ui.profile.RetrofitProfileProvider
+import org.wikiedufoundation.wikiedudashboard.util.Urls
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+
+/**
+ * Use the [apiModule] to creating Retrofit2 api service
+ * for WikiEduDashboardApi
+ *
+ * @return retrofit.create(WikiEduDashboardApi::class.java) */
+val apiModule = module {
+
+    single { provideBaseRetrofit().create(WikiEduDashboardApi::class.java) }
+
+    single { provideCommonsRetrofit().create(WikiEduDashboardMediaApi::class.java) }
+
+}
+
+/**
+ * Use the [providerGson] to provide Gson
+ * @return  GsonBuilder*/
+fun providerGson(): Gson =
+        GsonBuilder()
+                .setLenient()
+                .create()
+
+/**
+ * Use the [provideInterceptor] to provide a HttpLoggingInterceptor
+ * @return  HttpLoggingInterceptor*/
+fun provideInterceptor(): HttpLoggingInterceptor {
+    val interceptor = HttpLoggingInterceptor()
+    interceptor.level = HttpLoggingInterceptor.Level.BODY
+    return interceptor
+}
+
+/**
+ * Use the [provideClient] to provide a OkHttpClient
+ * @return  OkHttpClient*/
+fun provideClient(): OkHttpClient =
+        OkHttpClient.Builder()
+                .addInterceptor(provideInterceptor())
+                .connectTimeout(5, TimeUnit.MINUTES)
+                .readTimeout(5, TimeUnit.MINUTES)
+                .build()
+
+/**
+ * Use the [provideBaseRetrofit] to provide a Retrofit with WITH_BASE_URL instance
+ * @return  Retrofit*/
+fun provideBaseRetrofit(): Retrofit =
+        Retrofit.Builder()
+                .baseUrl(Urls.BASE_URL)
+                .client(provideClient())
+                .addConverterFactory(GsonConverterFactory.create(providerGson()))
+                .build()
+
+/**
+ * Use the [provideCommonsRetrofit] to provide a Retrofit with WIKI_MEDIA_COMMONS instance
+ * @return  Retrofit*/
+fun provideCommonsRetrofit(): Retrofit =
+        Retrofit.Builder()
+                .baseUrl(Urls.WIKIMEDIA_COMMONS_BASE_URL)
+                .client(provideClient())
+                .addConverterFactory(GsonConverterFactory.create(providerGson()))
+                .build()
```

Comment:
If possible let's have it in another file as it is not related to DI.

Reply:
Okay ,I will place this in its original file

Reply:
I am adding this back in apiModule.kt

Reply:
In which file and which package?